src/lib.rs

```diff
@@ -50,6 +54,41 @@ static STORAGE_UID: AtomicUsize = ATOMIC_USIZE_INIT;
 #[derive(Debug, PartialEq)]
 pub struct DeadComponentError;
 
+/// The error type which is returned from using 
+/// [`look_back`](struct.CursorItem.html#method.look_back) 
+/// and [`look_ahead`](struct.CursorItem.html#method.look_ahead) of 
+/// [`CursorItem`](struct.CursorItem.html).
+#[derive(Debug, PartialEq)]
+pub struct NotFoundError;
+
+/// The generic Error type of `froggy`.
+/// It can be either [`NotFoundError`](struct.NotFoundError.html)
+/// or [`DeadComponentError`](struct.DeadComponentError.html)
+pub struct Error(u8);
```

Comment:
Yep, no good reason as far as I can see. I'll change it to enum, I dunno why I choose `struct`